parser/GBenchToASV.py

```diff
@@ -0,0 +1,134 @@
+import os
+import sys
+import json
+import subprocess
+import argparse
+import re
+import platform
+import psutil
+
+from asvdb import ASVDb, BenchmarkInfo, BenchmarkResult
+
+# USAGE:
+#
+#   -d : JSON Result Directory
+#   -n : Repository Name
+#   -t : Target Directory for ASV JSON
+#   -b : Branch Name
+
+def build_argparse():
+    parser = argparse.ArgumentParser(add_help=True)
+    parser.add_argument('-d', nargs=1, help='JSON Result Directory')
+    parser.add_argument('-n', nargs=1, help='Repository Name')
+    parser.add_argument('-t', nargs=1, help='Target Directory for JSON')
+    parser.add_argument('-b', nargs=1, help='Branch Name')
+    return parser
+
+
+def getCommandOutput(cmd):
+    result = subprocess.run(cmd,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True)
+    stdout = result.stdout.decode().strip()
+    if result.returncode == 0:
+        return stdout
+
+    stderr = result.stderr.decode().strip()
+    raise RuntimeError("Problem running '%s' (STDOUT: '%s' STDERR: '%s')"
+                       % (cmd, stdout, stderr))
+
+
+def getSysInfo():
+    uname = platform.uname()
+    commitHash = getCommandOutput("git rev-parse HEAD")
+    commitTime = getCommandOutput("git log -n1 --pretty=%%ct %s" % commitHash)
+    commitTime = str(int(commitTime)*1000)  # ASV wants commit to be in milliseconds
+
+    bInfo = BenchmarkInfo(
+                machineName=uname.machine,
+                cudaVer=getCommandOutput("nvcc --version | grep release | awk '{print $5}' | tr -d ,"),
+                osType="%s %s" % (uname.system, uname.release),
+                pythonVer=platform.python_version(),
+                commitHash=commitHash,
+                commitTime=commitTime,
+                gpuType=getCommandOutput("nvidia-smi | awk '{print $3,$4}' | sed '8!d'"),
```

Comment:
This is probably fine but I know sometimes `nvidia-smi` will truncate the full name. Another way to get this is to use pynvml like this: https://github.com/rapidsai/benchmark/blob/3690ea966c7533badf29df7e572904f218d17bee/rapids_pytest_benchmark/rapids_pytest_benchmark/plugin.py#L466